permissions/src/iosMain/kotlin/BluetoothPermissionManager.kt

```diff
@@ -17,62 +18,95 @@ Copyright 2019 Splendo Consulting B.V. The Netherlands
-        return when (cbCentralManager.state.toLong()) {
-            STATE_UNKNOWN -> Support.NOT_SUPPORTED
-            STATE_RESETTING -> Support.RESETTING
-            STATE_UNSUPPORTED -> Support.NOT_SUPPORTED
-            STATE_UNAUTHORIZED -> Support.UNAUTHORIZED
-            STATE_POWERED_OFF -> Support.POWER_OFF
-            STATE_POWERED_ON -> Support.POWER_ON
-            else -> Support.NOT_SUPPORTED
+        return when (CBCentralManagerState.byCBManagerState(cbCentralManager.state)) {
+            CBCentralManagerState.UNSUPPORTED -> Support.NOT_SUPPORTED
+            CBCentralManagerState.RESETTING -> Support.RESETTING
+            CBCentralManagerState.UNKNOWN -> Support.NOT_SUPPORTED
+            CBCentralManagerState.UNAUTHORIZED -> Support.UNAUTHORIZED
+            CBCentralManagerState.POWER_OFF -> Support.POWER_OFF
+            CBCentralManagerState.POWER_ON -> Support.POWER_ON
         }
     }
 
     override suspend fun checkPermit(): Permit {
         //TODO: Add support for iOS 13. `authorizationStatus` is being deprecated
-        return when (cbPeripheralManager.authorizationStatus().toLong()) {
-            AUTHORIZATION_STATUS_NOT_DETERMINED -> Permit.UNDEFINED
-            AUTHORIZATION_STATUS_RESTRICTED -> Permit.RESTRICTED
-            AUTHORIZATION_STATUS_DENIED -> Permit.DENIED
-            AUTHORIZATION_STATUS_AUTHORIZED -> Permit.ALLOWED
-            else -> Permit.UNDEFINED
+        //https://github.com/splendo/kaluga/issues/21
+        return when (CBPeripheralManagerAuthorizationStatus.byCBPeripheralManagerAuthorizationStatus(authorizationStatusProvider())) {
+            CBPeripheralManagerAuthorizationStatus.CBPeripheralManagerAuthorizationStatusNotDetermined -> Permit.UNDEFINED
+            CBPeripheralManagerAuthorizationStatus.CBPeripheralManagerAuthorizationStatusRestricted -> Permit.RESTRICTED
+            CBPeripheralManagerAuthorizationStatus.CBPeripheralManagerAuthorizationStatusDenied -> Permit.DENIED
+            CBPeripheralManagerAuthorizationStatus.CBPeripheralManagerAuthorizationStatusAuthorized -> Permit.ALLOWED
         }
     }
 
-    companion object {
-        const val STATE_UNKNOWN = 0L
-        const val STATE_RESETTING = 1L
-        const val STATE_UNSUPPORTED = 2L
-        const val STATE_UNAUTHORIZED = 3L
-        const val STATE_POWERED_OFF = 4L
-        const val STATE_POWERED_ON = 5L
-
-        const val AUTHORIZATION_STATUS_NOT_DETERMINED = 0L
-        const val AUTHORIZATION_STATUS_RESTRICTED = 1L
-        const val AUTHORIZATION_STATUS_DENIED = 2L
-        const val AUTHORIZATION_STATUS_AUTHORIZED = 3L
+    //https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerstate
+    enum class CBCentralManagerState {
+        UNKNOWN,
+        RESETTING,
+        UNSUPPORTED,
+        UNAUTHORIZED,
+        POWER_OFF,
+        POWER_ON;
+
+        companion object {
+            fun byCBManagerState(state: CBManagerState): CBCentralManagerState {
```

Comment:
I made an extension method for this, but of course it's in Components now. Can you make an issue to update this later (when there is a "base" component or something), either for yourself or in the repo?